test_programs/execution_success/turbofish_call_func_diff_types/src/main.nr

```diff
@@ -0,0 +1,36 @@
+use dep::std::hash::Hasher;
+use dep::std::hash::poseidon2::Poseidon2Hasher;
+use dep::std::hash::poseidon::PoseidonHasher;
+
+fn main(x: Field, y: pub Field) {
+    let mut hasher = PoseidonHasher::default();
+    hasher.write(x);
+    hasher.write(y);
+    let poseidon_expected_hash = hasher.finish();
+    // Check that we get the same result when using the hasher in a 
+    // method that purely uses trait methods without a supplied implementation.
+    assert(hash_simple_array::<PoseidonHasher>([x, y]) == poseidon_expected_hash);
+
+    // Now let's do the same logic but with a different `Hasher` supplied to the turbofish operator
+    // We want to make sure that we have correctly monomorphized a function with a trait generic
+    // where the generic is not used on any function parameters or the return value.
+    let mut hasher = Poseidon2Hasher::default();
+    hasher.write(x);
+    hasher.write(y);
+    let poseidon2_expected_hash = hasher.finish();
+    assert(hash_simple_array::<Poseidon2Hasher>([x, y]) == poseidon2_expected_hash);
+}
+
+fn hash_simple_array<H>(input: [Field; 2]) -> Field where H: Hasher + Default {
+    // Check that we can a trait method to instead a trait implementation
+    // TODO: Need to remove the need for this type annotation
+    // TODO: Curently, without the annotation we will get `Expression type is ambiguous` when trying to use the `hasher`
+    let mut hasher: H = H::default();
```

Comment:
What type does `H::default()` return if not `H`? I'd assume an unbound type variable.

Still, this is suspicious since master does not have this issue.